search-sentiments-map/routes/trends.js

```diff
@@ -162,27 +163,70 @@ async function deleteAncientTrend() {
 }
 
 /** 
- * Finds the globally trending topics based on trending topics in each country.
- * Currently returning all topics from the US. TODO(@chenyuz): get a mixture of
- * topics from different countries.
+ * Deletes all trends entries in the Datastore. 
+ * Call this function when there is an update of the trends data structure. 
+ * TODO(@chenyuz): Delete this function when everything is done.
+ */
+async function deleteAllTrends() {
+  const query = datastore.createQuery('TrendsEntry').order('timestamp');
+  const [trendsEntries] = await datastore.runQuery(query);
+
+  for (let i = 0; i < trendsEntries.length; i++) {
+    const trendKey = trendsEntries[i][datastore.KEY];
+    await datastore.delete(trendKey);
+    console.log(`Trend ${trendsEntryKey.id} deleted.`)
+  }
+}
+
+/** 
+ * Finds the trending topics that appear the most across all recorded countries 
+ * and gets the top 10 as the globally trending topics.
  * @param {!Array<JSON>} trendsByCountry An array where each element is a country
  * and its trends.
- * @return {!Array<JSON>} Globally trending topics and their originating countries.
+ * @return {!Array<JSON>} 10 globally trending topics and the number of countries 
+ * where they are trending.
  */
 function getGlobalTrends(trendsByCountry) {
-  // Find all trends list(s) whose designated country is the US.
-  let UStrends = trendsByCountry.filter(trends => trends.country === 'US');
-  UStrends = UStrends[0].trends;
+  // Use a map to count the number of occurences of each trend.
+  let trendCountsMap = new Map();
+  trendsByCountry.forEach(trends => {
+    trends.trends.forEach(trend => {
+      let topic = trend.topic;
+      if (trendCountsMap.has(topic)) {
+        let newCount = trendCountsMap.get(topic) + 1;
+        trendCountsMap.set(topic, newCount);
+      } else {
+        trendCountsMap.set(topic, 1)
+      }
+    });
+  });
```

Comment:
Would it be possible to keep track of this as you add the trends in originally? So we only loop through them once.

Reply:
Yeah it is possible, but then we need to make the map a global variable, and it's actually the same asymptotic complexity, so I'm not sure.. @ntarn idea?